### PR TITLE
[Extension] Disable removing and updating system extension in Cloud Shell

### LINKED ISSUE #24023: Upgrade `ml` extension in Cloud Shell fails: FileExistsError: [Errno 17] File exists: '/usr/lib/python3.9/site-packages/azure-cli-extensions/ml'
## Describe the bug

**Command Name**
`az extension update`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
[Errno 17] File exists: '/usr/lib/python3.9/site-packages/azure-cli-extensions/ml'
Traceback (most recent call last):
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/extension/operations.py", line 413, in update_extension
    _add_whl_ext(cli_ctx=cmd_cli_ctx, source=download_url, ext_sha256=ext_sha256,
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/extension/operations.py", line 113, in _add_whl_ext
    raise CLIError('The extension {} already exists.'.format(extension_name))
knack.util.CLIError: The extension ml already exists.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/az/lib/python3.9/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/command_modules/extension/custom.py", line 34, in update_extension_cmd
    return update_extension(cli_ctx=cmd.cli_ctx, extension_name=extension_name, index_url=index_url,
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/extension/operations.py", line 421, in update_extension
    shutil.copytree(backup_dir, extension_path)
  File "/usr/lib/python3.9/shutil.py", line 568, in copytree
    return _copytree(entries=entries, src=src, dst=dst, symlinks=symlinks,
  File "/usr/lib/python3.9/shutil.py", line 467, in _copytree
    os.makedirs(dst, exist_ok=dirs_exist_ok)
  File "/usr/lib/python3.9/os.py", line 225, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/usr/lib/python3.9/site-packages/azure-cli-extensions/ml'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az extension update -n {}`

## Environment Summary
```
Linux-5.4.0-1091-azure-x86_64-with-glibc2.35 (Cloud Shell), CBL-Mariner/Linux
Python 3.9.13
Installer: RPM

azure-cli 2.40.0

Extensions:
ai-examples 0.2.5
ml 2.7.1
ssh 1.1.2

Dependencies:
msal 1.18.0b1
azure-mgmt-resource 21.1.0b1
```

### LINKED ISSUE #24405: `az extension update -n ssh` fails on Cloud Shell
## Describe the bug

**Command Name**
`az extension update`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
[Errno 17] File exists: '/usr/lib/python3.9/site-packages/azure-cli-extensions/ssh'
Traceback (most recent call last):
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/extension/operations.py", line 413, in update_extension
    _add_whl_ext(cli_ctx=cmd_cli_ctx, source=download_url, ext_sha256=ext_sha256,
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/extension/operations.py", line 113, in _add_whl_ext
    raise CLIError('The extension {} already exists.'.format(extension_name))
knack.util.CLIError: The extension ssh already exists.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/az/lib/python3.9/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/command_modules/extension/custom.py", line 34, in update_extension_cmd
    return update_extension(cli_ctx=cmd.cli_ctx, extension_name=extension_name, index_url=index_url,
  File "/usr/lib64/az/lib/python3.9/site-packages/azure/cli/core/extension/operations.py", line 421, in update_extension
    shutil.copytree(backup_dir, extension_path)
  File "/usr/lib/python3.9/shutil.py", line 568, in copytree
    return _copytree(entries=entries, src=src, dst=dst, symlinks=symlinks,
  File "/usr/lib/python3.9/shutil.py", line 467, in _copytree
    os.makedirs(dst, exist_ok=dirs_exist_ok)
  File "/usr/lib/python3.9/os.py", line 225, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/usr/lib/python3.9/site-packages/azure-cli-extensions/ssh'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az extension update -n {}`

## Environment Summary
```
Linux-5.4.0-1094-azure-x86_64-with-glibc2.35 (Cloud Shell), CBL-Mariner/Linux
Python 3.9.14
Installer: RPM

azure-cli 2.41.0

Extensions:
interactive 0.4.5
ai-examples 0.2.5
ml 2.10.0
ssh 1.1.2

Dependencies:
msal 1.20.0b1
azure-mgmt-resource 21.1.0b1
```

### PR DESCRIPTION
**Pull request by bebound**
**Related command**

`extension remove`
`extension update`

**Description**

Close #24023, close #24405

Add warning info that users cannot remove or update system extension in Cloud Shell, as they have no permission for `/usr/lib/python3.9/site-packages/azure-cli-extensions/`

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Extension

File: src/azure-cli-core/azure/cli/core/extension/operations.py [hunk 1]
```diff
@@ -353,6 +353,15 @@ def add_extension(cmd=None, source=None, extension_name=None, index_url=None, ye
         pass
 
 
+def is_cloud_shell_system_extension(ext_path):
+    import re
+    from azure.cli.core.util import in_cloud_console
+    if in_cloud_console():
+        if re.search(r'^/usr/lib/python3\.\d+/site-packages/azure-cli-extensions/', ext_path):
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/extension/operations.py:360]: Maybe check `startswith` `/usr/lib` should be enough. `re` is slow compared to `startswith`.

File: src/azure-cli-core/azure/cli/core/extension/operations.py [hunk 2]
```diff
@@ -391,6 +402,9 @@ def update_extension(cmd=None, extension_name=None, index_url=None, pip_extra_in
     try:
         cmd_cli_ctx = cli_ctx or cmd.cli_ctx
         ext = get_extension(extension_name, ext_type=WheelExtension)
+        if is_cloud_shell_system_extension(ext.path):
+            raise CLIError(
+                f"Cannot update system extension {extension_name}, please wait until Cloud Shell update it in the future release.")
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/extension/operations.py:407]:
```suggestion
                f"Cannot update system extension {extension_name}, please wait until Cloud Shell updates it in the next release.")
```

File: src/azure-cli-core/azure/cli/core/extension/operations.py [hunk 3]
```diff
@@ -361,6 +369,8 @@ def remove_extension(extension_name):
             raise CLIError(
                 "Extension '{name}' was installed in development mode. Remove using "
                 "`azdev extension remove {name}`".format(name=extension_name))
+        if is_cloud_shell_system_extension(ext.path):
+            raise CLIError(f"Cannot remove system extension {extension_name} in Cloud Shell.")
```
Comment (jsntcy) [inline src/azure-cli-core/azure/cli/core/extension/operations.py:373]: So is there a way for customer to remove the extension?

Comment (bebound) [reply src/azure-cli-core/azure/cli/core/extension/operations.py:373]: No, they don't have permission to remove system extension in Cloud Shell.

**Review by jsntcy — APPROVED** [review]

**Pull request merged**